### PR TITLE
fix light shape property causing incorrect light mapping

### LINKED ISSUE #2716: Sponza looks different with candidate 1.38.0 release
With 1.37.1:

image

https://playcanvas.com/editor/scene/423508

With latest master branch:

image

https://playcanvas.com/editor/scene/423508?use_local_engine=https://code.playcanvas.com/playcanvas-latest.dbg.js

Interestingly enough, there were already errors in the JS console for this scene. So there might be more than one problem in play here.

### PR DESCRIPTION
**Pull request by raytranuk**
Fixes #2716

this fixes light map problems with unmigrated lights in editor

I confirm I have signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

**Review by slimbuck — APPROVED** [review]

File: src/scene/light.js [hunk 1]
```diff
@@ -452,7 +452,7 @@ class Light {
     }
 
     get shape() {
-        return this._shape;
+        return this._shape ? this._shape : LIGHTSHAPE_PUNCTUAL;
```
Comment (willeastcott) [inline src/scene/light.js:455]: Surely this is unnecessary if `_shape` is always guaranteed to be set to a valid value.

**Comment by mvaligursky** [comment]
how about Light.updateKey() - it uses _shape as well

File: src/scene/light.js (hunk 1 above)
Comment (raytranuk) [reply src/scene/light.js:455]: I can test - but it doesn't seem that in the editor _shape will use class defaults if not present in the component - which is why editor migration code is needed

File: src/scene/light.js [hunk 2]
```diff
@@ -463,7 +463,7 @@ class Light {
             this.uploadAreaLightLUTs();
         }
 
-        this._shape = value;
+        this._shape = value ? value : LIGHTSHAPE_PUNCTUAL;
```
Comment (willeastcott) [inline src/scene/light.js:466]: Is this to guard against value being `null` or `undefined`? If so, won't it be undesirable to call `uploadAreaLightLUTs` above? Where are 'bad' values being passed to this function anyway?

File: src/scene/light.js (hunk 1 above)
Comment (raytranuk) [reply src/scene/light.js]: tested and removed the redundant code in the get accessor

File: src/scene/light.js (hunk 2 above)
Comment (raytranuk) [reply src/scene/light.js]: now using a sanitizedValue - I'm not sure where the bad values are coming from - I had assumed the _shape property was set once to the property default on construction - and left alone if not used - I can investigate further.

File: src/scene/light.js [hunk 3]
```diff
@@ -456,21 +456,22 @@ class Light {
     }
 
     set shape(value) {
```
Comment (mvaligursky) [inline src/scene/light.js:458]: just use default value

`set shape(value = LIGHTSHAPE_PUNCTUAL) {`

Comment (raytranuk) [reply src/scene/light.js]: done

File: src/scene/light.js (hunk 2 above)
Comment (raytranuk) [reply src/scene/light.js]: used default param for sanitizing value - also removed incorrect assignment in initializeComponentData()

**Review by mvaligursky — APPROVED** [review]
looks good now

**Pull request merged**